tools/integration_tests/inactive_stream_timeout/setup_test.go

```diff
@@ -0,0 +1,203 @@
+	expectedMsgSubstring := fmt.Sprintf("Closing reader for object %q due to inactivity.", objectName)
+
+	file, err := os.Open(logFile)
+	require.NoError(t, err, "Failed to open log file")
+	defer file.Close()
+
+	logLines, err := loadLogLines(file)
+	require.NoError(t, err, "Failed to read log file")
+
+	found := false
+	for _, line := range logLines {
+		logEntry, err := read_logs.ParseJsonLogLineIntoLogEntryStruct(line) // Assuming read_logs can parse general log lines too or a more generic parser is available.
+		// If parsing fails, it might be a non-JSON line or a different structured log.
+		// For this specific message, we expect it to be in the "Message" field of a structured log.
+
+		if err == nil && logEntry != nil {
+			// Check if the log entry's timestamp is within the expected window.
+			if (logEntry.Timestamp.After(startTime) || logEntry.Timestamp.Equal(startTime)) &&
+				(logEntry.Timestamp.Before(endTime) || logEntry.Timestamp.Equal(endTime)) {
+				if strings.Contains(logEntry.Message, expectedMsgSubstring) {
+					found = true
+					break
+				}
+			}
+		}
+	}
+
+	if shouldBePresent {
+		require.True(t, found, "Expected log message substring '%s' not found between %v and %v", expectedMsgSubstring, startTime, endTime)
+	} else {
+		require.False(t, found, "Unexpected log message substring '%s' found between %v and %v", expectedMsgSubstring, startTime, endTime)
+	}
+}
+
+func mountGCSFuseAndSetupTestDir(ctx context.Context, flags []string, storageClient *storage.Client, testDirName string) {
+	setup.MountGCSFuseWithGivenMountFunc(flags, gMountFunc)
+	setup.SetMntDir(gMountDir)
+	gTestDirPath = client.SetupTestDirectory(ctx, storageClient, testDirName)
+}
+
+// createConfigFile generate mount config.yaml.
+func createConfigFile(flags *gcsfuseTestFlags) string {
+	mountConfig := map[string]interface{}{
+		"read": map[string]interface{}{
+			"inactive-stream-timeout": flags.inactiveReadTimeout.String(),
+		},
+		"gcs-connection": map[string]interface{}{
+			"client-protocol": flags.clientProtocol,
+		},
+		"logging": map[string]interface{}{
+			"file-path": setup.LogFile(),
+			"format":    "json", // Ensure JSON logs for easier parsing
+		},
+	}
+	filePath := setup.YAMLConfigFile(mountConfig, flags.fileName)
+	return filePath
+}
+
+func TestMain(m *testing.M) {
+	setup.ParseSetUpFlags()
+
+	// Create common storage client to be used in test.
+	gCtx = context.Background()
+	closeStorageClient := client.CreateStorageClientWithCancel(&gCtx, &gStorageClient)
+	defer func() {
+		err := closeStorageClient()
+		if err != nil {
+			log.Fatalf("closeStorageClient failed: %v", err)
+		}
+	}()
+
+	setup.ExitWithFailureIfBothTestBucketAndMountedDirectoryFlagsAreNotSet()
+
+	if setup.MountedDirectory() != "" {
+		gMountDir = setup.MountedDirectory()
+		setup.SetLogFile(kLogFileNameForMountedDirectoryTests)
+		// Run tests for mounted directory if the flag is set.
+		os.Exit(m.Run())
```

Comment:
Why are we not using this [function](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/util/setup/setup.go#L355)?

Also where are we using  gMountDir?

Reply:
Can't use directly, as we are setting the log-file for mountedDir test.

gMountDir is used in every TestSetup `mountGCSFuseAndSetupTestDir` in setup_test.go file.